### DISCUSSION TITLE
Layout fields in form

### DISCUSSION THREAD

**Discussion by AndrewTraub** [data-driven-forms/react-forms#996] [Ideas]
I had started something like this and added a width variable to my fields so that two or three inputs could appear on the same line.  Any plans on doing this?  I'm happy to share my code that works with bootstrap's layout system.

**Comment by rvsia** [comment]
We do not provide any basic layout components because:

1. some design systems do not have any layout components 😬
2. writing custom layout components is simple as there is no field logic
3. some mappers provide basic layout options (for example, each MUI mapper component is already wrapped in Grid component and through that you can basically make anything you need)

However, I think some global `data-driven-forms/layouts` package could be a cool idea to implement and provide. 🤔  What do you think, @Hyperkid123 ?

**Reply by Hyperkid123** [reply]
Hi @AndrewTraub. We actually use to have component mapper split into form components and layout components in v1, but we have ultimately decided to remove the split and just use one component.

It was replaced by the FormTemplate and by adding the option of exact mapping props to each composite component.

> However, I think some global data-driven-forms/layouts package could be a cool idea to implement and provide.

As far as creating a new "layout" package. Sure it can be done I am just wondering how much use it's going to be. The reason why I've decided to remove the layout mapper is the fact that each application and even form in the app may have different layout requirements. So it's pretty much impossible to avoid some additional in-app customizations/overrides. To be clear I am not against it. It would basically be "just an extension of component mappers".

> I'm happy to share my code that works with bootstrap's layout system.

I think I'd like to see the code first because it may be something already existing in current mappers. In fact, very recently there was a similar question posted about the MUI mapper: https://github.com/data-driven-forms/react-forms/discussions/997 and an older one: https://github.com/data-driven-forms/react-forms/issues/871

I have not seen bootstrap for quite some time now so I am not familiar with the current API.

To propose a solution for you that you can use right now, I would create my own layout components and use them in a similar fashion as the `sub-form` component is (which is a remnant of the layout component mapper). https://data-driven-forms.org/mappers/sub-form?mapper=mui#heading-muisubform

Sorry this was a long one :smile:
